src/gcp/apphosting.ts

```diff
@@ -334,6 +347,33 @@ export async function getBuild(
   return res.body;
 }
 
+/**
+ * List Builds by backend
+ */
+export async function listBuilds(
+  projectId: string,
+  location: string,
+  backendId: string,
+): Promise<ListBuildsResponse> {
+  const name = `projects/${projectId}/locations/${location}/backends/${backendId}/builds`;
+  let pageToken: string | undefined = undefined;
+  const res: ListBuildsResponse = {
+    builds: [],
+    unreachable: [],
+  };
+
+  do {
+    const queryParams: Record<string, string> = pageToken ? { pageToken } : {};
+    const int = await client.get<ListBuildsResponse>(name, { queryParams });
+    res.builds.splice(res.builds.length, 0, ...(int.body.builds || []));
+    res.unreachable?.splice(res.unreachable.length, 0, ...(int.body.unreachable || []));
```

Comment:
Is it cleaner to use `.push`?

```suggestion
    res.builds.push(...(int.body.builds || []));
    res.unreachable?.push(...(int.body.unreachable || []));
```